OpenRA.Mods.Common/Widgets/Logic/Ingame/GameTimerLogic.cs

```diff
@@ -37,15 +37,15 @@ public GameTimerLogic(Widget widget, OrderManager orderManager, World world)
 				if (world.Timestep == 1)
 					return "Max Speed";
 
-				return "{0}% Speed".F(world.LobbyInfo.GlobalSettings.Timestep * 100 / world.Timestep);
+				return "{0}% Speed".F(world.OriginalTimestep * 100 / world.Timestep);
 			};
 
 			if (timer != null)
 			{
 				// Timers in replays should be synced to the effective game time, not the playback time.
 				var timestep = world.Timestep;
 				if (world.IsReplay)
-					timestep = world.WorldActor.Trait<MapOptions>().GameSpeed.Timestep;
+					timestep = world.OriginalTimestep;
```

Comment:
This is one of the cases I was referring to above: we can use `world.OriginalTimestep` directly for both replays and non-replays below (or, even better, `world.Timestep` after fixing the replay speed implementation).

Reply:
This is pretty much solved regardless of whether we go for a replay speed modifier or replay-exclusive timestep, so resolving here to improve overview.